src/promptflow/promptflow/_core/operation_context.py

```diff
@@ -133,10 +119,14 @@ def append_user_agent(self, user_agent: str):
         Args:
             user_agent (str): The user agent information to append.
         """
-        agent = self.get(self._USER_AGENT, '')
-        user_agent = user_agent.strip()
-        if user_agent not in agent:
-            self[self._USER_AGENT] = f"{agent} {user_agent}".strip()
+        agents = self.get('user_agent', '').split(' ')
+        user_agents = set(user_agent.split(' '))
+        distinct_agents = []
+        for user_agent in user_agents:
+            if user_agent not in agents and user_agent != '':
+                distinct_agents.append(user_agent)
+
+        self['user_agent'] = ' '.join(agents).join(distinct_agents)
```

Comment:
Does this line behave as expected? I think you are join two list together. Will the first part 
```
' '.join(agents)
```
return a new join string as next part `.join(distinct_agents)`

That is ' '.join(['string1', 'string2', 'string3']) .join(["a", "b", "c"])
will be like "string1 string2 string3".join(["a", "b", "c"])
then generate "astring1 string2 string3bstring1 string2 string3c"

Do you have test case to prove it correctness?